test/builder/pipeline.go

```diff
@@ -0,0 +1,185 @@
+)
+
+// PipelineOp is an operation which modify a Pipeline struct.
+type PipelineOp func(*v1alpha1.Pipeline)
+
+// PipelineSpecOp is an operation which modify a PipelineSpec struct.
+type PipelineSpecOp func(*v1alpha1.PipelineSpec)
+
+// PipelineTaskOp is an operation which modify a PipelineTask struct.
+type PipelineTaskOp func(*v1alpha1.PipelineTask)
+
+// PipelineRunOp is an operation which modify a PipelineRun struct.
+type PipelineRunOp func(*v1alpha1.PipelineRun)
+
+// PipelineRunSpecOp is an operation which modify a PipelineRunSpec struct.
+type PipelineRunSpecOp func(*v1alpha1.PipelineRunSpec)
+
+// PipelineResourceOp is an operation which modify a PipelineResource struct.
+type PipelineResourceOp func(*v1alpha1.PipelineResource)
+
+// PipelineResourceSpecOp is an operation which modify a PipelineResourceSpec struct.
+type PipelineResourceSpecOp func(*v1alpha1.PipelineResourceSpec)
+
+// Pipeline creates a Pipeline with default values.
+// Any number of Pipeline modifier can be passed to transform it.
+func Pipeline(name string, ops ...PipelineOp) *v1alpha1.Pipeline {
+	p := &v1alpha1.Pipeline{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: defaultNamespace,
+			Name:      name,
+		},
+	}
+
+	for _, op := range ops {
+		op(p)
+	}
+
+	return p
+}
+
+// PipelineNamespace sets the namespace of the Pipeline
+func PipelineNamespace(namespace string) PipelineOp {
+	return func(p *v1alpha1.Pipeline) {
+		p.ObjectMeta.Namespace = namespace
+	}
+}
+
+// PipelineSpec sets the PipelineSpec to the Pipeline.
+// Any number of PipelineSpec modifier can be passed to transform it.
+func PipelineSpec(ops ...PipelineSpecOp) PipelineOp {
+	return func(p *v1alpha1.Pipeline) {
+		ps := &p.Spec
+
+		for _, op := range ops {
+			op(ps)
+		}
+
+		p.Spec = *ps
+	}
+}
+
+// PipelineTask adds a PipelineTask, with specified name and task name, to the PipelineSpec.
+// Any number of PipelineTask modifier can be passed to transform it.
+func PipelineTask(name, taskName string, ops ...PipelineTaskOp) PipelineSpecOp {
+	return func(ps *v1alpha1.PipelineSpec) {
+		pTask := &v1alpha1.PipelineTask{
+			Name: name,
+			TaskRef: v1alpha1.TaskRef{
+				Name: taskName,
+			},
+		}
+		for _, op := range ops {
+			op(pTask)
+		}
+		ps.Tasks = append(ps.Tasks, *pTask)
+	}
+}
+
+// PipelineRun creates a PipelineRun with default values.
+// Any number of PipelineRun modifier can be passed to transform it.
+func PipelineRun(name string, ops ...PipelineRunOp) *v1alpha1.PipelineRun {
+	pr := &v1alpha1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: defaultNamespace,
+			Name:      name,
+		},
+		Spec: v1alpha1.PipelineRunSpec{
+			PipelineTriggerRef: v1alpha1.PipelineTriggerRef{
+				Type: v1alpha1.PipelineTriggerTypeManual,
+			},
+		},
+	}
+
+	for _, op := range ops {
+		op(pr)
+	}
+
+	return pr
+}
+
+// PipelineRunNamespace sets the namespace of the PipelineRun.
+func PipelineRunNamespace(namespace string) PipelineRunOp {
```

Comment:
We wouldn't need this function if `PipelineRun` function is updated to above suggestion.